src/parser/Parser.ts

```diff
@@ -1313,17 +1328,24 @@ export class Parser {
         this.warnIfNotBrighterScriptMode('namespace');
         let keyword = this.advance();
 
-        if (!this.isAtRootLevel()) {
-            this.diagnostics.push({
-                ...DiagnosticMessages.keywordMustBeDeclaredAtRootLevel('namespace'),
-                range: keyword.range
-            });
+        const parentNamespaces: NamespaceStatement[] = [];
+
+        for (const scope of this.parserScopeStack) {
+            if (scope instanceof NamespaceScope) {
+                parentNamespaces.push(scope.namespaceStatement);
+            } else {
+                this.diagnostics.push({
+                    ...DiagnosticMessages.keywordMustBeDeclaredAtNamespaceLevel('namespace'),
+                    range: keyword.range
+                });
```

Comment:
Move this validation check into the `BrsFileValidator` class, in the `NamespaceStatement visitor. (here-ish https://github.com/rokucommunity/brighterscript/blob/e8ae1d95e25298d2d629d9a73249968ab43b76fe/src/bscPlugin/validation/BrsFileValidator.ts#L103-L109)

The validation should walk up the .parent chain and add the diagnostic if a namespace has any parents that aren't also namespaces (or Body also I think).